### PR TITLE
mokutil: init at 0.5.0

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/mokutil/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    sha256 = "sha256-dt41TCr6RkmE9H+NN8LWv3ogGsK38JtLjVN/b2mbGJs=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ openssl efivar keyutils ];
+
+  meta = {
+    homepage = "https://github.com/lcp/mokutil";
+    description = "The utility to manipulate machines owner keys which managed in shim";
+    license = lib.licenses.gpl3Only;
```
Comment (papojari) [inline pkgs/tools/security/mokutil/default.nix:19]:
```suggestion
    license = lib.licenses.gpl3Plus;
```
The source code says GPL v3 or any later version. https://github.com/lcp/mokutil/blob/master/src/keyring.c

File: pkgs/tools/security/mokutil/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    maintainers = with lib.maintainers; [ nickcao ];
```
Comment (papojari) [inline pkgs/tools/security/mokutil/default.nix:20]:
```suggestion
  meta = with lib; {
    homepage = "https://github.com/lcp/mokutil";
    description = "The utility to manipulate machines owner keys which managed in shim";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ nickcao ];
```

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 163434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mokutil</li>
  </ul>
</details>

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/mokutil/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, lib, fetchFromGitHub, autoreconfHook, pkg-config, openssl, efivar, keyutils }:
+stdenv.mkDerivation rec {
```
Comment (papojari) [inline pkgs/tools/security/mokutil/default.nix:2]:
```suggestion

stdenv.mkDerivation rec {
```

**Review by auroraanna — APPROVED** [review]

File: pkgs/tools/security/mokutil/default.nix (hunk 3 above)
Comment (papojari) [reply pkgs/tools/security/mokutil/default.nix:2]: meant to request changes not approve

File: pkgs/tools/security/mokutil/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    rev = version;
+    sha256 = "sha256-dt41TCr6RkmE9H+NN8LWv3ogGsK38JtLjVN/b2mbGJs=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ openssl efivar keyutils ];
+
+  meta = with lib;{
+    homepage = "https://github.com/lcp/mokutil";
+    description = "The utility to manipulate machines owner keys which managed in shim";
```
Comment (papojari) [inline pkgs/tools/security/mokutil/default.nix:19]:
```suggestion
    description = "Utility to manipulate machines owner keys";
```
This is shorter and it's also the same as the description in the man page.

**Review by auroraanna — CHANGES_REQUESTED** [review]
Also this works for all platforms, right?

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 163434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mokutil</li>
  </ul>
</details>

**Review by auroraanna — APPROVED** [review]

**Comment by papojari** [comment]
##### Reviewed points

- [ ] package path fits guidelines
- [ ] package name fits guidelines
- [ ] package version fits guidelines
- [ ] package build on ARCHITECTURE
- [ ] executables tested on ARCHITECTURE
- [ ] `meta.description` is set and fits guidelines
- [ ] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [ ] `meta.maintainers` is set
- [ ] build time only dependencies are declared in `nativeBuildInputs`
- [ ] source is fetched using the appropriate function
- [ ] phases are respected

**Comment by bobby285271** [comment]
@ofborg build mokutil

**Pull request merged**